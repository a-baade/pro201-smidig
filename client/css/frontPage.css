/* Style for frontPage.jsx */

.front-image {
    display: flex;
    justify-content: flex-end;
    align-items: center;
    flex-direction: column;
    height: 90vh;
    background-position: top;
}

.front-image-text {
    text-align: center;
    font-size: 64px;
    text-shadow: 0 0 10px #333232;
    color: white;
    margin-bottom: 10vh;
}

.front-image-button {
    z-index: 99;
    font-size: 14px;
    font-weight: 500;
    text-transform: uppercase;
    letter-spacing: 2px;
    background-color: #8AB7BA;
    border-radius: 10px;
    border: 0;
    min-width: 150px;
    width: 20vw;
    padding: 15px 30px;
    margin-bottom: 25vh;
}
.about-us-title {
    text-align: center;
}

.about-us-div {
    position: relative;
    margin: 80px;
    padding: 12px 25px;
    padding-inline: 0.5em;
    background-color: #95BDC4;
    border-radius: 5px;
    box-sizing: border-box;
    text-align: left;
    line-height: 25px;
    font-size: 18px;
}
.about-us-div h1 {
    margin-bottom: 5vh;
}

.about-us-div p {
    padding: 35px;
}

.about-us-button {
    position: relative;
    left: 45%;
    margin: 3vh 10px 10px 10px;
    padding: 17px 32px;
    border-radius: 10px;
    border: 0;
    background: #d5e5e6;
    text-transform: uppercase;
    letter-spacing: 2px;
    font-size: 12px;
    font-weight: 600;

}

.topic-container {
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    gap: 50px;
    text-align: center;
    margin: 80px;
}

.topic h4 {
    padding: 10px;
    font-weight: bold;
    font-size: 27px;
}

.topic-image img {
    width: 300px;
    height: 300px;
    border-radius: 50%;
}

.topic p {
    width: 100%;
    min-width: 320px;
    max-width: 400px;
}

.newsfeed-container {
    background-color: #95BDC4;
    margin: 80px;
    padding: 12px 20px;
    padding-inline: 0.5em;
    box-sizing: border-box;
    border-radius: 5px;
    text-align: center;
    font-size: 20px;
    line-height: 25px;
}

.newsfeed-items-container {
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    gap: 10px;
    margin: 80px 40px;
    box-sizing: border-box;
    text-align: center;
    font-size: 14px;
}

.newsfeed-item {
    box-sizing: content-box;
    background-color: white;
    border-radius: 5px;
    padding-bottom: 10px;
    width: 30%;
    min-width: 250px;
    height: auto;
    min-height: 300px;
    display: flex;
    flex-direction: column;
}

.newsfeed-image {
    object-fit: cover;
    width: 100%;
    height: 250px;
    border-top-left-radius: 5px;
    border-top-right-radius: 5px;
}

.newsfeed-item p {
    margin: 5px;
    text-align: left;
    padding: 15px;
}

.read-more {
    display: flex;
    height: 100%;
    width: 100%;
    justify-content: center;
    align-items: end;
    font-weight: bold;
    text-decoration: underline;
    text-align: center;
    cursor: pointer;
}