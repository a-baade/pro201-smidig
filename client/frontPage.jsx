
import React from "react";


export function FrontPage() {

    return (
      <div>
       <h1>Front page</h1>
      </div>
    );
}
