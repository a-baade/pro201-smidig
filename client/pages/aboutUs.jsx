import React from "react";
import MainBg from "../pics/about-us-pictures/main-bg.png";
import FirstLayerArt from "../pics/about-us-pictures/firstLayerArt.png";
import TopLayerArt from "../pics/about-us-pictures/topLayerArt.png";
import PovertyFirst from "../pics/about-us-pictures/povertyFirstLayer.png";
import PovertySecond from "../pics/about-us-pictures/povertytopLayer.png";
import ZeroHunger from "../pics/about-us-pictures/zeroHungerFirst.png";
import ZeroHungerTop from "../pics/about-us-pictures/seroHungerTop.png";
import GoodHealthFirst from "../pics/about-us-pictures/goodHealthFirst.png";
import GoodHealthTop from "../pics/about-us-pictures/goodHealthTop.png";
import QualityEducation from "../pics/about-us-pictures/qualityEducation.png";
import QualityEducationTop from "../pics/about-us-pictures/qualityEducationTop.png";
import GenderEquality from "../pics/about-us-pictures/genderEqualityFirst.png";
import GenderEqualityTop from "../pics/about-us-pictures/genderEqualityTop.png";

export function AboutUs() {
  return (
    <div>
      <div>
        <img src={MainBg} className={"main-bg-pic"} alt={"mainBg"}/>
      </div>
      <div className={"second-row-box"}>
        <div className={"knowledge-text-box"}>
          <p className={"our-mission"}>OUR MISSION</p>
          <h1 className={"knowledge-power"}>KNOWLEDGE = POWER</h1>
          <p>We're a Norwegian female-founded startup changing corporate giving and the way companies and nonprofits connect.</p>
          <p>We're saving companies resources, help create stable income streams for nonprofits, and create deeper relationships between donor and nonprofit.</p>
          <p> We help implement solid strategies for impactful and sustainable giving, all in line with the UN SDGs.</p>
        </div>
        <div className={"art-box"}>
          <img src={FirstLayerArt} className={"first-layer-art"} alt={"firstLayer"}/>
          <img src={TopLayerArt} className={"top-layer-art"} alt={"topLayer"}/>
          <p className={"our-mission mission"}>OUR MISSION</p>
          <h1 className={"action-call"}>A CALL TO ACTION</h1>
        </div>
      </div>
      <div className={"big-container"}>
        <div className={"small-box light"}>
          <p className={"small-title"}>No poverty</p>
          <p className={"small-txt"}>We are committed to end poverty in all its forms everywhere</p>
          <img src={PovertyFirst} className={"poverty-first"}/>
          <img src={PovertySecond} className={"poverty-second"}/>
        </div>
        <div className={"small-box dark"}>
          <p className={"small-title"}>Zero hunger</p>
          <p className={"small-txt"}>End hunger, achieve food security and improve nutrition and promote sustainable agriculture</p>
          <img src={ZeroHunger} className={"zero-hunger"}/>
          <img src={ZeroHungerTop} className={"zero-hunger-top"}/>
        </div>
        <div className={"small-box light"}>
          <img src={GoodHealthFirst} className={"good-health"}/>
          <img src={GoodHealthTop} className={"good-health-top"}/>
          <p className={"small-title good-health-title"}>Good health</p>
          <p className={"small-txt good-health-txt"}>Ensure healthy lives and promote well-being for all at all ages</p>
        </div>
        <div className={"small-box light"}>
          <p className={"small-title"}>Quality education</p>
          <p className={"small-txt"}>Ensure inclusive and equitable quality education and promote lifelong learning opportunities for all</p>
          <img src={QualityEducation} className={"quality-education"}/>
          <img src={QualityEducationTop} className={"quality-education-top"}/>
        </div>
        <div className={"small-box dark"}>
          <img src={GenderEquality} className={"gender-equality"}/>
          <img src={GenderEqualityTop} className={"gender-equality-top"}/>
          <p className={"small-title gender-eql-title"}>Gender equality</p>
          <p className={"small-txt gender-eql-txt"}>Achieve gender equality and empower all women and girls</p>
        </div>
        <div className={"small-box light"}>
          <p className={"small-title"}>Clean water & sanitation</p>
          <p className={"small-txt"}>Ensure availability and sustainable management of water and sanitation for all</p>
          <img src={ZeroHunger} className={"zero-hunger"}/>
          <img src={ZeroHungerTop} className={"zero-hunger-top"}/>
        </div>
        <div className={"small-box light"}>
          <p className={"small-title"}>Affordable & clean energy</p>
          <p className={"small-txt"}>Ensure access to affordable, reliable, sustainable and modern energy for all</p>
          <img src={PovertyFirst} className={"poverty-first"}/>
          <img src={PovertySecond} className={"poverty-second"}/>
        </div>
        <div className={"small-box light"}>
          <p className={"small-title"}>Decent work & economic growth</p>
          <p className={"small-txt"}>Promote sustained, inclusive and sustainable economic growth, full and productive employment and decent work for all</p>
          <img src={ZeroHunger} className={"decent-work"}/>
          <img src={ZeroHungerTop} className={"decent-work-top"}/>
        </div>
        <div className={"small-box light"}>
          <p className={"small-title"}>Industry, innovation & infrastructure</p>
          <p className={"small-txt"}>Build resilient infrastructure, promote inclusive and sustainable industrialization and foster innovation</p>
          <img src={PovertyFirst} className={"industry"}/>
          <img src={PovertySecond} className={"industry-second"}/>
        </div>
        <div className={"small-box dark"}>
          <p className={"small-title"}>Reduce inequalities</p>
          <p className={"small-txt"}>Reduce inequality within and among countries</p>
          <img src={ZeroHunger} className={"zero-hunger"}/>
          <img src={ZeroHungerTop} className={"zero-hunger-top"}/>
        </div>
        <div className={"small-box light"}>
          <p className={"small-title"}>Subsainable cities & communinites</p>
          <p className={"small-txt"}>Make cities and human settlements inclusive, safe, resilient and sustainable</p>
          <img src={PovertyFirst} className={"poverty-first"}/>
          <img src={PovertySecond} className={"poverty-second"}/>
        </div>
        <div className={"small-box dark"}>
          <p className={"small-title"}>responsible consumption & production</p>
          <p className={"small-txt"}>Ensure sustainable consumption and production patterns</p>
          <img src={ZeroHunger} className={"zero-hunger"}/>
          <img src={ZeroHungerTop} className={"zero-hunger-top"}/>
        </div>
        <div className={"small-box light"}>
          <p className={"small-title"}>Climate action</p>
          <p className={"small-txt"}>Take urgent action to combat climate change and its impacts</p>
          <img src={PovertyFirst} className={"poverty-first"}/>
          <img src={PovertySecond} className={"poverty-second"}/>
        </div>
        <div className={"small-box dark"}>
          <p className={"small-title"}>Life below water</p>
          <p className={"small-txt"}>Conserve and sustainably use the oceans, seas and marine resources for sustainable development</p>
          <img src={ZeroHunger} className={"zero-hunger"}/>
          <img src={ZeroHungerTop} className={"zero-hunger-top"}/>
        </div>
        <div className={"small-box light"}>
          <p className={"small-title"}>Life on land</p>
          <p className={"small-txt"}>Protect, restore and promote sustainable use of terrestrial ecosystems, sustainably manage forests, combat desertification, and halt and reverse land degradation and halt biodiversity loss</p>
          <img src={PovertyFirst} className={"life-first"}/>
          <img src={PovertySecond} className={"life-second"}/>
        </div>
        <div className={"small-box dark"}>
          <p className={"small-title"}>Peace, justice & strong institution</p>
          <p className={"small-txt"}>Promote peaceful and inclusive societies for sustainable development, provide access to justice for all and build effective, accountable and inclusive institutions at all levels</p>
          <img src={ZeroHunger} className={"peace-first"}/>
          <img src={ZeroHungerTop} className={"peace-top"}/>
        </div>
        <div className={"small-box light"}>
          <p className={"small-title"}>Partnerships for the goals</p>
          <p className={"small-txt"}>Strengthen the means of implementation and revitalize the global partnership for sustainable development</p>
          <img src={PovertyFirst} className={"partnership-first"}/>
          <img src={PovertySecond} className={"partnership-second"}/>
        </div>

        <div className={"small-box dark"}>
          <img src={ZeroHunger} className={"un-goals-first"}/>
          <img src={ZeroHungerTop} className={"un-goals-top"}/>
          <p className={"un-goals-title"}>THE UN SDG GOALS</p>
        </div>
      </div>
    </div>
  );
}