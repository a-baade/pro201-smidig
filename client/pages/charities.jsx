import React, { useContext, useEffect, useState } from "react";
import { Link } from "react-router-dom";
import { ApiContext } from "../apiContext";
import { useLoading } from "../useLoading";
import { fetchJSON } from "../lib/fetchJSON";
import LoadingWheel from "../components/loadingSpinner";

function CharityCard({
  charity: { _id, name, description, bgImage, charityLogo },
}) {
  // Charity card with button to donate or charity page
  return (
    <div className={"charity-card custom-shadow"}>
      <div
        className={"charity-background"}
        style={{ backgroundImage: `url( data:image/png;base64${bgImage} )` }}
      >
        <img
          src={`data:image/png;base64${charityLogo}`}
          className={"charity-img custom-shadow"}
        />
      </div>

      <div className={"charity-card-content"}>
        <h3>{name}</h3>
        <p>{description}</p>
        <div className={"card-buttons"}>
          <Link to={{ pathname: "/charities/charity/id?id=" + _id }}>
            <button className="card-button custom-shadow-button">
              Read more
            </button>
          </Link>
          <Link to={{ pathname: "/charities/donate/id?id=" + _id }}>
            <button className="card-button custom-shadow-button">Donate</button>
          </Link>
        </div>
      </div>
    </div>
  );
}

export function Charities() {
  const { allCharities } = useContext(ApiContext);
  const { category, setCategory } = useState("");
  const { categoryQuery, setCategoryQuery } = useState("");
  const [search, setSearch] = useState("");
  const [data, setData] = useState([]);
  const { loading, error } = useLoading(
    async () => await allCharities({ category }),
    [category]
  );

  useEffect(async () => {
    if (search === "") {
      await fetchJSON("/api/charities").then((jsonData) => {
        setData(jsonData);
      });
    } else {
      await fetchJSON(`/api/charities/search/?name=${search}`).then(
        (jsonData) => {
          setData(jsonData);
        }
      );
    }
  }, [search]);

  function handleSubmitQuery(e) {
    e.preventDefault();
    setCategory(categoryQuery);
  }

  async function handleSearch(event) {
    await setSearch(event);
  }

  if (loading) {
    return <LoadingWheel />;
  }

  if (error) {
    return (
      <div>
        <h1>Error</h1>
        <div id="error-text">{error.toString()}</div>
      </div>
    );
  }

  // Add category filter
  return (
    <div className={"page-margin"}>
      <div className="container-charities">
        <div className="search-charities">
          <input
            className="search-input-charities"
            placeholder="Search Charities"
            value={search}
            onChange={(e) => handleSearch(e.target.value)}
          />
          {search && (
            <div className={"charity-results"}>Results for {search}</div>
          )}
        </div>
        <div></div>

        {/*<form onSubmit={handleSubmitQuery}>*/}
        {/*    <button  id={"category-query"}*/}
        {/*             value={categoryQuery}*/}
        {/*             onChange={(e) => setCategoryQuery(e.target.value)}>Water</button>*/}
        {/*    <button  id={"category-query"}*/}
        {/*             value={categoryQuery}*/}
        {/*             onChange={(e) => setCategoryQuery(e.target.value)}>Knowledge</button>*/}
        {/*</form>*/}
      </div>
      <div>
        <ul className={"charity-cards-container"}>
          {data?.map((charity, index) => (
            <CharityCard key={index} charity={charity} />
          ))}
        </ul>
      </div>
    </div>
  );
}
