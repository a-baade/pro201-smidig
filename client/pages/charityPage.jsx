import React, { useContext } from "react";
import { Link, useSearchParams} from "react-router-dom";
import { ApiContext } from "../apiContext";
import { useLoading } from "../useLoading";
import Arrow from "../pics/icons/arrow.png";
import NewsImg1 from "../pics/bg-pictures/img_3.png";
import NewsImg2 from "../pics/bg-pictures/img_4.png";
import NewsImg3 from "../pics/bg-pictures/img_5.jpg";
import Facebook from "../pics/icons/facebook.png";
import Instagram from "../pics/icons/instagram.png";
import Snapchat from "../pics/icons/snapchat.png";
import Tiktok from "../pics/icons/tiktok.png";
import WWW from "../pics/icons/world-wide-web.png";
import LoadingWheel from "../components/loadingSpinner";

function Charity({charity:
    { _id, name, description, bgImage, charityLogo }
}) {

  // Charity page
  return (
    <div>
      <Link to={"/charities"}>
        <div className={"back-button custom-shadow-button"}>
          <img src={Arrow} />
        </div>
      </Link>
      <img
        src={`data:image/png;base64${bgImage}`}
        className={"charity-page-background"}
      />
      <div className={"charity-page-content"}>
        <div className={"charity-page-description"}>
          <div className={"charity-description-top"}>
            <img
              src={`data:image/png;base64${charityLogo}`}
              className={"charity-description-logo"}
            />
            <h2>{name}</h2>
          </div>
          <div className={"charity-description-text"}>
            <p>{description}</p>
          </div>
        </div>
        <div className={"charity-page-side"}>
          <Link to={{ pathname: "/charities/donate/id?id=" + _id }}>
            <button className={"charity-page-donate-button custom-shadow-button"}>Donate</button>
          </Link>
          <a
            href={"https://www.google.com/"}
            className={"charity-social-links"}
          >
            <img src={WWW} /> examplecharity.com
          </a>
          <a
            href={"https://www.facebook.com/"}
            className={"charity-social-links"}
          >
            <img src={Facebook} /> Example Charity
          </a>
          <a
            href={"https://www.instagram.com/"}
            className={"charity-social-links"}
          >
            <img src={Instagram} /> @examplecharity
          </a>
          <a
            href={"https://www.tiktok.com/"}
            className={"charity-social-links"}
          >
            <img src={Tiktok} /> @examplecharity
          </a>
          <a
            href={"https://www.snapchat.com/"}
            className={"charity-social-links"}
          >
            <img src={Snapchat} /> examplecharity
          </a>
        </div>
      </div>
      <div className={"newsfeed-container custom-shadow"}>
        <h1>NEWSFEED</h1>
        <div className={"newsfeed-items-container"}>
          <div className={"newsfeed-item custom-shadow"}>
            <img
                src={NewsImg1}
                className={"newsfeed-image"}
                alt={"Afghan children in school"}
            />
            <p>
              Afghan refugee children enrolling in norwegian nurseries 2 weeks
              after coming to norway. Prime minister hopes that these efforts
              will help stabilize the everyday life of the children and that
              they will integrate quicker.
            </p>
            <p className={"read-more"}>Read more</p>
          </div>
          <div className={"newsfeed-item custom-shadow"}>
            <img
                src={NewsImg2}
                className={"newsfeed-image"}
                alt={"Deers in nature"}
            />
            <p>
              Hunting season has ended, yet people are still hunting animals
              illegally. Protecting nature is our responsibility, and make sure
              that the next generation can enjoy wilderness like us.
            </p>
            <p className={"read-more"}>Read more</p>
          </div>
          <div className={"newsfeed-item custom-shadow"}>
            <img
                src={NewsImg3}
                className={"newsfeed-image"}
                alt={"Akeem in karate training"}
            />
            <p>
              Thanks to the support of charities, children like Akeem are able
              to partake in after school activities. Children like Akeem would
              not be able to afford ...
            </p>
            <p className={"read-more"}>Read more</p>
          </div>
        </div>
      </div>
    </div>
  );
}

export default function CharityPage() {
  let [searchParams, setSearchParams] = useSearchParams();
  const id = searchParams.get("id");

  const {oneCharity} = useContext(ApiContext);
  const {loading, error, data} = useLoading(
    async () => await oneCharity(id),
    []
  );

  if (loading) {
    return <LoadingWheel/>;;
  }

  if (error) {
    return (
      <div>
        <h1>Error</h1>
        <div id="error-text">{error.toString()}</div>
      </div>
    );
  }

  return (
    <div className={"page-margin"}>
      <ul className={"charity-page-container"}>
        {data.map((charity, index) => (
          <Charity key={index} charity={charity} />
        ))}
      </ul>
    </div>
  );
}
