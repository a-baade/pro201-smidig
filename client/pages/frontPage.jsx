import React from "react";
import FrontBg from "../pics/bg-pictures/water-pic.png";
import Nature from "../pics/bg-pictures/img.png";
import Welfare from "../pics/bg-pictures/img_1.png";
import Education from "../pics/bg-pictures/img_2.png";
import NewsImg1 from "../pics/bg-pictures/img_3.png";
import NewsImg2 from "../pics/bg-pictures/img_4.png";
import NewsImg3 from "../pics/bg-pictures/img_5.jpg";
import { Link } from "react-router-dom";

export function FrontPage() {
  return (
    <div className={"page-margin"}>
      <div
        className={"front-image"}
        style={{ backgroundImage: `url(${FrontBg})` }}
      >
        <h1 className={"front-image-text custom-font-knowledge"}>
          Meliora Impact
        </h1>
        <Link to={"/signup"}>
          <button className={"front-image-button custom-shadow-button"}>
            Join us
          </button>
        </Link>
      </div>

      <div className={"about-us-div"}>
        <div className={"about-us-title"}>
          <h1>About us</h1>
        </div>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur
          viverra, eros at luctus molestie, dolor risus convallis orci, ut
          tempus tellus nunc quis nulla. Maecenas libero lectus, pharetra ut
          magna in, rutrum luctus lorem. Nulla blandit augue nulla, sit amet
          porta dolor interdum eu. Phasellus suscipit porttitor urna, non
          vestibulum dui volutpat quis. Donec viverra lectus diam, sed
          sollicitudin quam pretium at. Curabitur nisl ligula, porta nec finibus
          eu, venenatis vitae quam. Curabitur facilisis magna condimentum
          maximus feugiat. Aliquam erat volutpat. Mauris posuere vel mi non
          commodo. Nunc dapibus libero ipsum, placerat posuere augue vulputate
          vitae. Aenean volutpat sollicitudin accumsan. Vestibulum at maximus
          purus, in hendrerit eros. Phasellus id tellus a eros volutpat rutrum
          et in ligula. Donec quis odio augue. Mauris pellentesque feugiat lacus
          eget ornare.
        </p>
        <Link to={"/about"}>
          <button className={"about-us-button custom-shadow-button"}>
            Read more
          </button>
        </Link>
      </div>
      <div className={"topic-container"}>
        <div className={"topic"}>
          <h4>Protecting nature</h4>
          <div className={"topic-image"}>
            <img src={Nature} alt={"Picture of nature"} />
          </div>
          <p>
            Nature is by design sustainable, it has supported itself for decades
            but human intervention across the past centuries has been weakening
            the very infrastructure it stands on. We always aim to protect and
            invest in what is already here so we as well it stands a chance for
            future growth.
          </p>
        </div>
        <div className={"topic"}>
          <h4>Welfare</h4>
          <div className={"topic-image"}>
            <img src={Welfare} alt={"Picture of poor children"} />
          </div>
          <p>
            The clean water crisis persist as a leading issue in several large
            parts of the world. It is a human right and necessity yet is
            consistently overlooked and lacking in support. Our aim is to uplift
            and highlight these core violations of human rights to foster more
            support for the communities at large.
          </p>
        </div>
        <div className={"topic"}>
          <h4>Education</h4>
          <div className={"topic-image"}>
            <img src={Education} alt={"Picture of child in school"} />
          </div>
          <p>
            Studies prove that education plays a large role in how well people
            in communities of all ranges fare supporting themselves and breaking
            cycles generational poverty and trauma. We strongly believe it is a
            right to have access to quality education and learning which is why
            it is part of our focus.
          </p>
        </div>
      </div>
      <div className={"newsfeed-container custom-shadow"}>
        <h1>NEWSFEED</h1>
        <div className={"newsfeed-items-container"}>
          <div className={"newsfeed-item custom-shadow"}>
            <img
              src={NewsImg1}
              className={"newsfeed-image"}
              alt={"Afghan children in school"}
            />
            <p>
              Afghan refugee children enrolling in norwegian nurseries 2 weeks
              after coming to norway. Prime minister hopes that these efforts
              will help stabilize the everyday life of the children and that
              they will integrate quicker.
            </p>
            <p className={"read-more"}>Read more</p>
          </div>
          <div className={"newsfeed-item custom-shadow"}>
            <img
              src={NewsImg2}
              className={"newsfeed-image"}
              alt={"Deers in nature"}
            />
            <p>
              Hunting season has ended, yet people are still hunting animals
              illegally. Protecting nature is our responsibility, and make sure
              that the next generation can enjoy wilderness like us.
            </p>
            <p className={"read-more"}>Read more</p>
          </div>
          <div className={"newsfeed-item custom-shadow"}>
            <img
              src={NewsImg3}
              className={"newsfeed-image"}
              alt={"Akeem in karate training"}
            />
            <p>
              Thanks to the support of charities, children like Akeem are able
              to partake in after school activities. Children like Akeem would
              not be able to afford ...
            </p>
            <p className={"read-more"}>Read more</p>
          </div>
        </div>
      </div>
    </div>
  );
}
