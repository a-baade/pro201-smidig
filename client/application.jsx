import { BrowserRouter, Link, Route, Routes } from "react-router-dom";
import { FrontPage } from "./pages/frontPage";
import React, { useContext } from "react";
import { LoginPage } from "./pages/loginPage";
import { useLoading } from "./useLoading";
import { ApiContext } from "./apiContext";
import { Profile } from "./pages/profile";
import { Charities } from "./pages/charities";
import Navigation from "./components/navigation";
import Footer from "./components/footer";
import { RegisterNewUser } from "./pages/registerNewUser";
import { AboutUs } from "./pages/aboutUs";
import { SignUp } from "./pages/signUp";
import CharityPage from "./pages/charityPage";
import { Contact } from "./pages/contact";
import { Dashboard } from "./pages/dashboard";
import Donate from "./pages/donate";
import LoadingWheel from "./components/loadingSpinner";

export function Application() {
  const { fetchLogin } = useContext(ApiContext);
  const { data, error, loading, reload } = useLoading(fetchLogin);

  if (error) {
    return <div>Error: {error.toString()}</div>;
  }
  if (loading) {
    return <LoadingWheel />;
  }

  return (
    <BrowserRouter>
      <header>
        <Navigation />
      </header>
      <main>
        <Routes>
          <Route path={"/"} element={<FrontPage />} />

          <Route path={"/charities"} element={<Charities />} />
          <Route path={"/charities/charity/:id"} element={<CharityPage />} />
          <Route path={"/charities/donate/:id"} element={<Donate />} />
          <Route path={"/register"} element={<RegisterNewUser />} />
          <Route path={"/about"} element={<AboutUs />} />
          <Route path={"/contact"} element={<Contact />} />
          <Route path={"/signup"} element={<SignUp config={data.config} />} />
          <Route
            path={"/dashboard"}
            element={<Dashboard user={data?.user} />}
          />

          <Route
            path={"/login/*"}
            element={<LoginPage config={data.config} reload={reload} />}
          />
          <Route
            path={"/signup/*"}
            element={<LoginPage config={data.config} reload={reload} />}
          />
          <Route path={"/profile"} element={<Profile user={data?.user} />} />
          <Route path={"*"} element={<h1>Not found</h1>} />
        </Routes>
      </main>
      <Footer />
    </BrowserRouter>
  );
}
