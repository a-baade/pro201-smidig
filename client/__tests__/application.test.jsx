import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter, Route, Routes } from "react-router-dom";
import { Application } from "../application";
import { FrontPage } from "../pages/frontPage";

describe("Application component", () => {
  it("should test div element", async function() {
    const divEl = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <Application/>
      </MemoryRouter>,
      divEl
    );
    expect(divEl.innerHTML).toMatchSnapshot();
  });

  it("should test header", async function() {
    const header = document.createElement("header");
    ReactDOM.render(
      <MemoryRouter>
        <Application/>
      </MemoryRouter>,
      header
    );
    expect(header.innerHTML).toMatchSnapshot();
  });

  it("should test Route", async function() {
    const route = document.createElement("Route");
    ReactDOM.render(
      <MemoryRouter>
        <Routes>
          <Route path={"/"} element={<FrontPage />} />
        </Routes>
      </MemoryRouter>,
      route
    );
    expect(route.innerHTML).toMatchSnapshot();
  });
})