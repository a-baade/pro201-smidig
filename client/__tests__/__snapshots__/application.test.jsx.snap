// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application component should test Route 1`] = `"<div class=\\"page-margin\\"><div class=\\"front-image\\"><h1 class=\\"front-image-text custom-font-knowledge\\">Meliora Impact</h1><a href=\\"/signup\\"><button class=\\"front-image-button custom-shadow-button\\">Join us</button></a></div><div class=\\"about-us-div\\"><div class=\\"about-us-title\\"><h1>About us</h1></div><p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur viverra, eros at luctus molestie, dolor risus convallis orci, ut tempus tellus nunc quis nulla. Maecenas libero lectus, pharetra ut magna in, rutrum luctus lorem. Nulla blandit augue nulla, sit amet porta dolor interdum eu. Phasellus suscipit porttitor urna, non vestibulum dui volutpat quis. Donec viverra lectus diam, sed sollicitudin quam pretium at. Curabitur nisl ligula, porta nec finibus eu, venenatis vitae quam. Curabitur facilisis magna condimentum maximus feugiat. Aliquam erat volutpat. Mauris posuere vel mi non commodo. Nunc dapibus libero ipsum, placerat posuere augue vulputate vitae. Aenean volutpat sollicitudin accumsan. Vestibulum at maximus purus, in hendrerit eros. Phasellus id tellus a eros volutpat rutrum et in ligula. Donec quis odio augue. Mauris pellentesque feugiat lacus eget ornare.</p><a href=\\"/about\\"><button class=\\"about-us-button custom-shadow-button\\">Read more</button></a></div><div class=\\"topic-container\\"><div class=\\"topic\\"><h4>Protecting nature</h4><div class=\\"topic-image\\"><img src=\\"[object Object]\\" alt=\\"Picture of nature\\"></div><p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce viverra id metus ut scelerisque. Morbi luctus tincidunt neque, sed sagittis ligula consectetur nec. Vivamus consequat neque at varius faucibus.</p></div><div class=\\"topic\\"><h4>Welfare</h4><div class=\\"topic-image\\"><img src=\\"[object Object]\\" alt=\\"Picture of poor children\\"></div><p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce viverra id metus ut scelerisque. Morbi luctus tincidunt neque, sed sagittis ligula consectetur nec. Vivamus consequat neque at varius faucibus.</p></div><div class=\\"topic\\"><h4>Education</h4><div class=\\"topic-image\\"><img src=\\"[object Object]\\" alt=\\"Picture of child in school\\"></div><p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce viverra id metus ut scelerisque. Morbi luctus tincidunt neque, sed sagittis ligula consectetur nec. Vivamus consequat neque at varius faucibus.</p></div></div><div class=\\"newsfeed-container custom-shadow\\"><h1>NEWSFEED</h1><div class=\\"newsfeed-items-container\\"><div class=\\"newsfeed-item custom-shadow\\"><img src=\\"[object Object]\\" class=\\"newsfeed-image\\" alt=\\"Afghan children in school\\"><p>Afghan refugee children enrolling in norwegian nurseries 2 weeks after coming to norway. Prime minister hopes that these efforts will help stabilize the everyday life of the children and that they will integrate quicker.</p><p class=\\"read-more\\">Read more</p></div><div class=\\"newsfeed-item custom-shadow\\"><img src=\\"[object Object]\\" class=\\"newsfeed-image\\" alt=\\"Deers in nature\\"><p>Hunting season has ended, yet people are still hunting animals illegally. Protecting nature is our responsibility, and make sure that the next generation can enjoy wilderness like us.</p><p class=\\"read-more\\">Read more</p></div><div class=\\"newsfeed-item custom-shadow\\"><img src=\\"[object Object]\\" class=\\"newsfeed-image\\" alt=\\"Akeem in karate training\\"><p>Thanks to the support of charities, children like Akeem are able to partake in after school activities. Children like Akeem would not be able to afford ...</p><p class=\\"read-more\\">Read more</p></div></div></div></div>"`;

exports[`Application component should test div element 1`] = `"<div class=\\"loading-wrapper\\"><div class=\\"loading\\"><div></div><div></div><div></div><div></div><div></div><div></div><div></div><div></div></div></div>"`;

exports[`Application component should test header 1`] = `"<div class=\\"loading-wrapper\\"><div class=\\"loading\\"><div></div><div></div><div></div><div></div><div></div><div></div><div></div><div></div></div></div>"`;
